FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,51 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard self != nil else { return }
+			switch result {
+			case .failure:
+				completion(.failure(Error.connectivity))
+			case .success(let response):
+				if response.1.statusCode != 200 {
+					completion(.failure(Error.invalidData))
+				} else {
+					completion(FeedLoaderDecoder.feedLoaderItemsDecoder(with: response.0))
+				}
+			}
+		}
+	}
+}
+
+fileprivate struct FeedLoaderDecoder {
+	private struct FeedLoaderResult: Decodable {
+		let items: [FeedImageItem]
+	}
+
+	private struct FeedImageItem: Decodable {
+		let image_id: String
+		let image_desc: String?
+		let image_loc: String?
+		let image_url: String
+
+		var item: FeedImage? {
+			guard let url = URL(string: image_url), let uuid = UUID(uuidString: image_id) else {
+				return nil
+			}
+			return FeedImage(id: uuid, description: image_desc, location: image_loc, url: url)
+		}
+	}
+
+	static func feedLoaderItemsDecoder(with data: Data) -> Swift.Result<[FeedImage], Error> {
+		do {
+			let decoder = JSONDecoder()
+			let feedLoader = try decoder.decode(FeedLoaderResult.self, from: data)
+			return Swift.Result.success(feedLoader.items.compactMap {
+				$0.item
+			}
+			)
+		} catch {
+			return Swift.Result.failure(RemoteFeedLoader.Error.invalidData)
+		}
```

Comment:
Since the error being `thrown` by `decode(_: from:)` is being ignored completely, it's possible to replace the `do-catch` block with an `if let` statement with an optional `try?` for the success case.
Also, there's no need to write `Swift.Result` for the return type, the Type System can infer that.
```suggestion
		let decoder = JSONDecoder()
		if let feedLoader = try? decoder.decode(FeedLoaderResult.self, from: data) {
			return .success(feedLoader.items.compactMap {
				$0.item
			})
		}
		return .failure(RemoteFeedLoader.Error.invalidData)
```

Reply:
done